pkg/lib/parentwebsocket.js

```diff
@@ -0,0 +1,45 @@
+import { transport_origin } from './location';
```

Comment:
Adding all these little internal cockpit.js helpers like location.ts or parentwebsocket.js annoys me. Before they were internal implementation details of cockpit.js, and now they become publicly accessible and available API for all cockpit projects. These are really not meant to be at the same level as e.g. `service.js`, and they also make it much harder for cockpit app developers to see which ones are "real" APIs worth exploring, and which ones are just internal noise.

Can we please move these to pkg/cockpit-api/ or something? They really shouldn't be in pkg/lib/. Or if pkg/lib/ is somehow special, then at least pkg/lib/cockpit-api/, with a README.md that says "this is not the library you are looking for`<jedi wave />`"

Reply:
I was thinking the same thing as I was doing this, but where do we draw the line?  I was imagining a directory called `_internal/`.  Does that work?

Reply:
If you strongly prefer `pkg/lib/_internal/`, I can live with that. I'd rather split it out by what uses it, though -- we may get similar helpers for e.g. shared PF components.

As external projects bundle cockpit.js, I suppose they have to live in pkg/lib --- so `pk/cockpit-api/` doesn't work.

Reply:
Done.